pypeit/specobjs.py

```diff
@@ -254,11 +266,20 @@ def get_std(self, multi_spec_det=None):
                 SNR = np.median(self.BOX_COUNTS*np.sqrt(self.BOX_COUNTS_IVAR), axis=1)
             else:
                 return None
+
+            # TODO: Hack this for now, but revisit once mosaicing has been
+            # implemented for DEIMOS.  I.e., does it make sense to keep this?
+            if multi_spec_det is not None and isinstance(self.DETECTOR[0], Mosaic):
+                msc = [d.det for d in self.DETECTOR[0].detectors]
+                _multi_spec_det = None if multi_spec_det == msc else multi_spec_det
+            else:
+                _multi_spec_det = multi_spec_det
+
             # For multiple detectors grab the requested detectors
-            if multi_spec_det is not None:
+            if _multi_spec_det is not None:
```

Comment:
I think the idea here is for a spec1d file which contains the standard on multiple detectors, to return only the spectrum of the standard for all three detectors. In this case it was being returned as a specobjs object with e.g. 3 spectra for gemini GMOS. I think that once mosaics work for GMOS, flux calibration will be performed with the mosaiced spectra and so this functionality may not be required anymore.

Reply:
I've changed how this is done in a way that should throw an error if you're using `multi_spec_det` with a mosaic.  We'll probably need to return to this in later PRs.